cmd/fleetctl/mdm.go

```diff
@@ -122,13 +122,9 @@ func mdmRunCommand() *cli.Command {
 				hostUUIDs = append(hostUUIDs, host.UUID)
 			}
 
-			if len(hostUUIDs) == 0 {
-				// all hosts were not found
-				return errors.New("No hosts targeted. Make sure you provide a valid hostname, UUID, osquery host ID, or node key.")
-			}
-			if notFoundCount > 0 {
-				// at least one was not found
-				return errors.New("One or more targeted hosts don't exist. Make sure you provide a valid hostname, UUID, osquery host ID, or node key.")
+			if len(hostUUIDs) == 0 || notFoundCount > 0 {
```

Comment:
Not something to fix now, but we should rely on `notFoundCount` only? (I can see one `continue` in the loop only and it has `notFoundCount++`)

Reply:
Had the same thought initially – dependence on `len(hostUUIDs)` covers the case where no hosts were provided to the `--hosts` flag (since `notFoundCount` would be 0 in that case). Chatted with @marko-lisica and confirmed we do want the same message in both cases.